include/mpsl/mpsl_work.h

```diff
@@ -42,13 +42,55 @@ extern struct k_work_q mpsl_work_q;
  *
  * @param work Address of work item.
  *
- * @return N/A
+ * @retval 0 if work was already submitted to a queue
+ * @retval 1 if work was not submitted and has been queued to @p queue
+ * @retval 2 if work was running and has been queued to the queue that was
+ * running it
+ * @retval -EBUSY
+ * * if work submission was rejected because the work item is cancelling; or
+ * * @p queue is draining; or
+ * * @p queue is plugged.
+ * @retval -EINVAL if @p queue is null and the work item has never been run.
+ * @retval -ENODEV if @p queue has not been started.
  */
 static inline int mpsl_work_submit(struct k_work *work)
```

Comment:
It's maybe a bit simpler if this and `mpsl_work_schedule` return void and assert inside the functions themselves. Then we don't need to repeat the zephyr docs for `k_work_submit_to_queue()` and `k_work_schedule_for_queue()`

Reply:
Or do as in zephyr does with `k_work_submit`, just refer to `k_work_submit_to_queue`. https://github.com/zephyrproject-rtos/zephyr/blob/1cfe90d825a1e07d9aa77bf389338dc72dfda6fe/include/zephyr/kernel.h#L3336-L3344

Reply:
I would prefer to keep asserting on the return value, can shorten the description like Herman's suggestion. Like this we take responsibility of it here, and can easily handle different return values if needed in the future. 
(If there are strong opinions I don't mind going with other solutions as well though.)

Reply:
I see, then I misunderstood it. Seems then I am overvoted in my weak opinion. Will change it next time I update the PR.

Reply:
If we find there's a need to implement some kind of error code handling we can always change it later